trie/verkle.go

```diff
@@ -230,33 +228,22 @@ func nodeToDBKey(n verkle.VerkleNode) []byte {
 // Commit writes all nodes to the trie's memory database, tracking the internal
 // and external (for account tries) references.
 func (trie *VerkleTrie) Commit(_ bool) (common.Hash, *NodeSet, error) {
-	flush := make(chan verkle.VerkleNode)
-	resolver := func(n verkle.VerkleNode) {
-		flush <- n
+	root, ok := trie.root.(*verkle.InternalNode)
+	if !ok {
+		return common.Hash{}, nil, errors.New("unexpected root node type")
+	}
+	nodes, err := root.BatchSerialize()
+	if err != nil {
+		return common.Hash{}, nil, fmt.Errorf("serializing tree nodes: %s", err)
 	}
-	go func() {
-		switch root := trie.root.(type) {
-		case *verkle.InternalNode:
-			root.Flush(resolver)
-		case *verkle.StatelessNode:
-			root.Flush(resolver)
-		}
-		close(flush)
-	}()
-	var commitCount int
-	for n := range flush {
-		commitCount += 1
-		value, err := n.Serialize()
-		if err != nil {
-			panic(err)
-		}
 
-		if err := trie.db.diskdb.Put(nodeToDBKey(n), value); err != nil {
-			return common.Hash{}, NewNodeSet(common.Hash{}), err
+	for _, node := range nodes {
+		if err := trie.db.diskdb.Put(node.CommitmentBytes[:], node.SerializedBytes); err != nil {
```

Comment:
I noticed that the `DiskDB()` API was removed. Not sure why, but I think that was a good change... is an open pandora box to let "anyone" jump on top of the trie layer. (Maybe was done for that reason).

Reply:
it's part of the preparations for this new database schema, iirc that's more or less what you assumed.